docs/deprecation.md

```diff
@@ -0,0 +1,39 @@
+# Deprecation Policy
+
+In order for any software to be reliable enough for production use, it needs to have a stable API. This means that
+users would have a high confidence that the way they use Kubevirt would not become obsolete, or at least would give
+the proper time to prepare if it does.
+
+This document aims to specify a process for several aspects of deprecation revolving Kubevirt. This is intended to
+be a "live document" which updates and refines itself over time.
+
+## Process
+
+### Feature Gates
+#### Introduction
+A feature gate should be introduced for every new feature, capability, configuration or any other change that
+is not yet stable and can jeopardize the cluster.
+
+A feature gate serves as:
+* A way to warn the user about a possible danger that could be caused by an unstable logic.
+* A way for the users to explicitly state that they would like to take the risk and enable the feature.
+
+Over time the new features would stabilize and graduate their API versions (i.e. from beta, to alpha to stable). When
+that happens, and therefore there risk of jeopardizing the user decreases significantly, the feature should be enabled
+by default and the feature gate should be deprecated (see section below).
+
+#### Deprecation
+When a feature stabilizes and there is no need to guard it by a feature gate, the feature gate should be deprecated.
```

Comment:
I think there is also another motivation and that is feature/API did not show as useful. I think we don't want to maintain unnecessary something that is not used.